frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield.jsx

```diff
@@ -331,6 +331,25 @@ export default class ExpressionEditorTextfield extends React.Component {
     });
   }
 
+  errorAsMarkers(errorMessage = null) {
+    if (errorMessage) {
+      const { pos, len } = errorMessage;
+      if (typeof pos === "number") {
```

Comment:
What are the situations in which `pos` is not a number? Does some part of the tokenizer/parser/compiler not support pos+length or is it a semantic limitation?

Maybe a comment explaining the special case?

Reply:
Only the tokenizer right now includes `pos` and `len`. Until we add the same information to all other modules, unfortunately we have to filter it here.